pkg/utils/tracing/span.go

```diff
@@ -1,51 +0,0 @@
-package tracing
-
-import (
-	"os"
-	"time"
-
-	"k8s.io/klog/v2/klogr"
-)
-
-/*
-	Poor Mans OpenTracing.
-
-	Standardizes logging of operation duration.
-*/
-
-var enabled = false
-var logger = klogr.New()
-
-func init() {
-	enabled = os.Getenv("ARGOCD_TRACING_ENABLED") == "1"
-}
-
-type Span struct {
-	operationName string
-	baggage       map[string]interface{}
-	start         time.Time
-}
-
-func (s Span) Finish() {
-	if enabled {
-		logger.WithValues(baggageToVals(s.baggage)).
```

Comment:
A bug! `WithValues()` accepts a list of key-value pairs but here code passes a slice of interfaces as the key and does not pass any value. Did it even work? The right thing to do here is to expand the slice using `...`, which is what I did above in the other file. `s.logger.WithValues(baggageToVals(s.baggage)...)`